pkg/auto/wordpress/job/energy.go

```diff
@@ -0,0 +1,101 @@
+package job
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+)
+
+// EnergyJob gets energy consumed over the previous interval (typically 24 hours)
+type EnergyJob struct {
+	BaseJob
+	client     gen.MeterHistoryClient
+	infoClient gen.MeterInfoClient
+	Meters     []string
+	Interval   time.Duration
+}
+
+func (e *EnergyJob) GetName() string {
+	return "energy"
+}
+
+func (e *EnergyJob) GetClients() []any {
+	return []any{&e.client, &e.infoClient}
+}
+
+func (e *EnergyJob) GetInterval() time.Duration {
+	return e.Interval
+}
+
+func (e *EnergyJob) Do(ctx context.Context, sendFn sender) error {
+	consumption := float32(.0)
+
+	now := time.Now()
+
+	for _, meter := range e.Meters {
+		multiplier, err := e.getUnitMultiplier(ctx, meter)
+
+		if err != nil {
+			e.Logger.Error("getting unit multiplier", zap.String("meter", meter), zap.Error(err))
+			continue
+		}
+
+		records, err := getAllRecords(ctx, e.client.ListMeterReadingHistory, meter, now, e.GetInterval())
+
+		if err != nil {
+			e.Logger.Error("getting all records", zap.String("meter", meter), zap.Error(err))
+			continue
+		}
+
+		consumption += processMeterRecords(multiplier, records)
```

Comment:
This _get all, then process_ is pretty inefficient wrt memory considering you only really need the first/last record within the range to compute the consumption.

Reply:
This is true. I wasn't familiar with the database API in terms of meter sampling frequency. Ideally, I just submit 2 ranges and take first and last for each range and subtract, or is there an easier way to do first/last in the listMeterReading call?

Reply:
There isn't, unfortunately. However you can get away with only keeping one page in memory at a time instead of all the records in range.